echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/microsoftteams/MicrosoftTeamsService.java

```diff
@@ -16,18 +16,64 @@
 
 package com.netflix.spinnaker.echo.microsoftteams;
 
+import com.netflix.spinnaker.retrofit.Slf4jRetrofitLogger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import lombok.extern.slf4j.Slf4j;
+import retrofit.RestAdapter;
+import retrofit.client.Client;
 import retrofit.client.Response;
+import retrofit.converter.JacksonConverter;
 
 @Slf4j
 public class MicrosoftTeamsService {
-  private final MicrosoftTeamsClient microsoftTeamsClient;
+  private static final String BASE_URL_REGEX_PATTERN = "^(http|https)://.+?/";
```

Comment:
do we have to live with a regex or could we use a somewhat standard class to parse URLs?

If this is really easiest to go with, it would be nice to see some tests showing what exactly this is extracting from the regex.

Reply:
Removed regex and use `java.net.URL` instead.